tests/utils/chunks.py

```diff
@@ -43,3 +64,43 @@ def create_tool_call_chunk(
             ]
         }
     )
+
+
+def _check_sse_line(actual: str, expected: Union[str, dict]):
+    if isinstance(expected, str):
+        assert (
+            actual == expected
+        ), f"actual line != expected line: {actual!r} != {expected!r}"
+        return
+
+    assert actual.startswith("data: "), f"Invalid data SSE entry: {actual!r}"
+    actual = actual[len("data: ") :]
+
+    try:
+        actual_dict = json.loads(actual)
+    except json.JSONDecodeError:
+        raise AssertionError(f"Invalid JSON in data SSE entry: {actual!r}")
+    assert (
+        actual_dict == expected
+    ), f"actual json != expected json: {actual_dict!r} != {expected!r}"
+
+
+ExpectedSSEStream = Iterable[Union[str, dict]]
+
+
+def check_sse_stream(actual: Iterable[str], expected: ExpectedSSEStream):
+    expected = itertools.chain(expected, ["data: [DONE]"])
+    expected = itertools.chain.from_iterable((line, "") for line in expected)
+
+    sentinel = object()
+    for a_line, e_obj in itertools.zip_longest(
+        actual, expected, fillvalue=sentinel
+    ):
+        assert (
```

Comment:
And in this option you don't need to construct this message, pytest will do this for you, and will pull values of list like that:

```bash
def test_demonstration():
        x = [1, 2, 3]
        y = [3, 5, 6, 6]
>       assert len(x) == len(y)
E       assert 3 == 4
E        +  where 3 = len([1, 2, 3])
E        +  and   4 = len([3, 5, 6, 6])
```

Reply:
1. `Iterable` doesn't have `len`
2. This wouldn't work because these asserts are called from non-test modules. `pytest` [doesn't rewrite](https://pytest.org/en/7.4.x/how-to/assert.html#assertion-introspection-details) such asserts automatically. It's possible to make it work via [register_assert_rewrite](https://pytest.org/en/7.4.x/how-to/writing_plugins.html#assertion-rewriting), but it isn't worth the effort in my opinion (yet).